client/lib/plugins/sanitize-section-content.js

```diff
@@ -180,7 +210,17 @@ export const sanitizeSectionContent = content => {
 		}
 	}
 
-	// once done walking the DOM tree
+	// swap out any nodes that need replacements
+	replacements.forEach( ( [ node, newNode ] ) => {
+		let child;
+
+		while ( ( child = node.firstChild ) ) {
+			newNode.appendChild( child );
+		}
+
+		node.parentNode.replaceChild( newNode, node );
```

Comment:
A more general case would also need to move the node attributes. But we don't allow any attributes for header nodes anyway.

Reply:
Yeah I almost added this in after reading your comment but I figured that _if_ we want to start allowing attributes on the replaced elements we can do so at that time. No need to run through more code now when we know it's not happening. Thanks!